app/models/manageiq/providers/vmware/infra_manager/ovf_service.rb

```diff
@@ -109,27 +118,29 @@ def provision_options
   def save_action_options(action, overrides)
     return unless action == ResourceAction::PROVISION
 
-    action_options = options.fetch_path(:config_info, action.downcase.to_sym).with_indifferent_access
-    %w[resource_pool ems_folder host].each do |r|
-      next if action_options[r].blank?
-
-      action_options["#{r}_id"] = action_options.delete(r).split.first.to_i
-    end
-
+    action_options = options.fetch_path(:config_info, action.downcase.to_sym).slice(*CONFIG_OPTIONS_WHITELIST).with_indifferent_access
     action_options.deep_merge!(parse_dialog_options)
     action_options.deep_merge!(overrides)
+    validate_target_name(action_options)
 
     options[action_option_key(action)] = action_options
     save!
   end
 
+  def validate_target_name(options)
+    unless ovf_template.target_name_valid?(options[:vm_name])
+      _log.info("A target with name [#{options[:vm_name]}] already exists.")
+      options[:vm_name] = "#{options[:vm_name]}_#{Time.current.to_i}"
```

Comment:
The catalog item could be ordered many times. We need to ensure the target name does not exist yet otherwise the provider side provisioning will fail. Is there any better way to ensure a uniq target name?
cc @tinaafitz